tests/Core/Util/Common/PrepareForOutputTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Tests for the \PHP_CodeSniffer\Util\Common::prepareForOutput() method.
+ *
+ * @author  Bartosz Dziewoński <dziewonski@fastmail.fm>
+ * @license https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Tests\Core\Util\Common;
+
+use PHP_CodeSniffer\Util\Common;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Tests for the \PHP_CodeSniffer\Util\Common::prepareForOutput() method.
+ *
+ * @covers \PHP_CodeSniffer\Util\Common::prepareForOutput
+ */
+final class PrepareForOutputTest extends TestCase
+{
+
+
+    /**
+     * Test formatting whitespace characters, on anything other than Windows.
+     *
+     * @param string   $content     The content to prepare.
+     * @param string[] $exclude     A list of characters to leave invisible.
+     * @param string   $expected    Expected function output.
+     * @param string   $expectedWin Expected function output on Windows (unused in this test).
+     *
+     * @dataProvider dataPrepareForOutput
+     *
+     * @return void
+     */
+    public function testPrepareForOutput($content, $exclude, $expected, $expectedWin)
+    {
+        if (stripos(PHP_OS, 'WIN') === 0) {
+            $this->markTestSkipped('Non Windows only');
+        }
```

Comment:
This can be replaced with a `@requires     OS ^(?!WIN).*` tag.
The regex format and the `OS` category are supported on all PHPUnit versions we currently need.

Refs:
* https://phpunit.de/manual/4.8/en/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests.requires.tables.api
* https://docs.phpunit.de/en/9.6/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests-requires-tables-api